### DISCUSSION TITLE
best practices for parsing GTF files - gene_version field?

### DISCUSSION THREAD

**Discussion by jamesnemesh** [alexdobin/STAR#2040] [Q&A] [answered]
Hi!  A sort of general question about interpreting GTF files.  Some GTF files contain a gene_version field, and sometimes there are genes in the GTF that have multiple ENSG IDs but the same gene symbol, where each ID is tied to a different gene_version.  

```
chrX HAVANA gene 102599512 102714671  .  +  .    gene_id ENSG00000271147; gene_version 8; gene_type lncRNA; gene_name ARMCX5-GPRASP2; level 2; hgnc_id HGNC:42000; tag overlapping_locus; havana_gene OTTHUMG00000022059.4;
chrX HAVANA gene 102712495 102753530  .  +  .     gene_id ENSG00000286237; gene_version 1; gene_type protein_coding; gene_name ARMCX5-GPRASP2; level 2; tag overlapping_locus; havana_gene OTTHUMG00000193744.1;
```

Are both of these entries equally valid, or should only the most recent version of the gene record be considered for analysis?  In particular, an interesting case has come up where one version of this gene overlaps another gene (BHLHB9) on the same strand, leading to ambiguous assignments of those reads when both versions of ARMCX5-GPRASP2 are considered.  However, if only the most recent version is considered, reads are assigned unambiguously to BHLHB9 instead.  

<img width="1773" alt="image" src="https://github.com/alexdobin/STAR/assets/4561831/99cec4dc-ae11-4951-942b-e76b8b3b3177">

Is there general thought on how to deal with this ambiguity, or is it best practice to keep either the latest or both versions of the gene?  

Thanks in advance for any advice - I know this isn't really a STAR or STARsolo question, but it does impact the STARsolo output.

**Comment by alexdobin** [comment]
Hi @jamesnemesh 

I am not 100% sure (it's better to ask ENSEMBL/Gencode people), but I think the gene version is just used to track the change in gene annotations between different releases of annotations. In a given release, each gene can only have one version.

**Comment by jamesnemesh** [comment]
If each gene can have only one version, then how do you parse the two lines above, that come from the same GTF?  Do you consider these two truly different biological entities (genes) that happen to have the same gene symbol?

The source of this data is the cellranger source GTF file `refdata-cellranger-arc-GRCh38-2020-A/genes/genes.gtf.gz`.

**Comment by alexdobin** [comment, marked as answer]
Since they have different gene_id, I would consider them different genes.
They are "read-through" fusion genes that combine two normal genes, so they are not normal genes.

**Comment by jamesnemesh** [comment]
Hi!  I just wanted to wrap this up and say I think you're absolutely right.  The two ENSGs are distinct gene models that happen to have the same gene symbol.  The V8 ENSG happens to have 7 other previous versions as encoded by the `.1`, `.2` etc at the end of the ENSG ID, while the V1 only has a single version.  This is more clear if you use the Ensembl [history query tool](http://www.ensembl.org/Homo_sapiens/Tools/IDMapper?db=core), where it looks like the V1 gene now reports `No stable IDs mapped to the given IDs`, so perhaps it will be removed from a later build.

Filtering to one of these two ENSGs based on the version happens to work, but that's completely random and no basis for systematic decision.

Thanks again for your thoughts!
